src/store/book.js

```diff
@@ -0,0 +1,60 @@
+import api from '../plugins/axios';
+
+const state = () => ({
+  books: {}
+});
+
+const getters = {
+  allBook: state => state.books
+};
+
+const mutations = {
+  setBooks: (state, books) => (state.books = books)
+};
+
+const actions = {
+  // get BOOK data from API
+  fetchBook({ commit }) {
+    api
+      .get('/books')
+      .then(res => {
+        commit('setBooks', res.data);
```

Comment:
we should be check response data be for do any action to make sure properties of object exist
if (res) {
}